_posts/2021-02-19-26-rln-relay.md

```diff
@@ -0,0 +1,195 @@
+
+#### Economic-Incentive Spam protection
+Is it the end of the spam-protection journey!? shall we get along with the spammers!? Certainly not!
+Waku RLN-Relay gives us a p2p spam-protection method which:
+- is **efficient**  i.e., with no unreasonable computational, storage, memory, and bandwidth requirement! as such, it fits the network of **heterogeneous** peers. 
+- respects users **privacy** unlike reputation-based and centralized methods.
+- deploys **economic-incentives** to contain spammers' activity. Namely, there is a financial sacrifice for those who want to spam the system. How? follow along ...
+
+We devise a general rule to save everyone's life and that is 
+
+**No one can publish more than M messages per epoch without being financially charged!** 
+
+We set M to 1 for now, but keep in mind, it can be any arbitrary value.
+You may be thinking that it is too restrictive! one per epoch!? but don't worry, we set the epoch to a reasonable value so that it does not slow down the communication of innocent users but will make the life of spammers harder! Epoch here can be every second, as defined by UTC date-time +-20s.
+
+The remainder of this post is all about the story of how to enforce this limit on each user's messaging rate as well as how to impose the financial cost when the limit gets violated.  This brings us to the Rate Limiting Nullifiers and how we integrate this technique into Waku v2 (in specific the Waku Relay protocol) to protect our valuable users against spammers.
+
+# Technical Terms
+
+**Zero-knowledge proof**: Zero-knowledge proof (ZKP)[^14] allows a *prover* to show a *verifier* that they know something, without revealing what that something is. This means you can do the trust-minimized computation that is also privacy-preserving. As a basic example, instead of showing your ID when going to a bar you simply give them proof that you are over 18, without showing the doorman your id. In this write-up, by ZKP we essentially mean zkSNARK[^15] which is one of the many types of ZKPs. 
+
+**Threshold Secret Sharing Scheme**: (m,n) Threshold secret-sharing is a method by which you can split a secret value s into n pieces in a way that the secret s can be reconstructed by having m pieces (m <= n). The economic-incentive spam protection utilizes a (2,n) secret sharing realized by Shamir Secret Sharing Scheme[^13].
+
+# Overview: Economic-Incentive Spam protection through Rate Limiting Nullifiers
```

Comment:
I'd add a paragraph here with "Context" that briefly summarizes previous post and all the improvements that have been made on blockers since then. Proof time etc. This gives a sense of continuity